app/controllers/users_controller.rb

```diff
@@ -1,4 +1,5 @@
 class UsersController < ApplicationController
+  before_action :authenticate_user!
```

Comment:
- You have done well so far, but according to the project requirements, please ensure that a user is authenticated before accessing any of your controller actions. Please add this to your application_controller👍